lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -550,27 +550,31 @@ class DxilLowerCreateHandleForLib : public ModulePass {
     ResourceRegisterAllocator.GatherReservedRegisters(DM);
 
     // Remove unused resources.
-    DM.RemoveResourcesWithUnusedSymbols();
+    if (!DM.GetConsistentBindings())
+      bChanged |= DM.RemoveResourcesWithUnusedSymbols();
 
     unsigned newResources = DM.GetCBuffers().size() + DM.GetUAVs().size() +
                             DM.GetSRVs().size() + DM.GetSamplers().size();
-    bChanged = bChanged || (numResources != newResources);
+    bChanged |= numResources != newResources;
```

Comment:
```suggestion
```
No longer needed because `bChanged` was already updated with the return value of `RemoveResourcesWithUnusedSymbols`?